src/models/Organism.tsx

```diff
@@ -1,64 +1,113 @@
-import { Substance, CellPart } from '../Types';
-import { cloneDeep } from 'lodash';
+import { types } from 'mobx-state-tree';
+import { v4 as uuid } from 'uuid';
+import { Organelle, OrganelleType } from './Organelle';
+import { SubstanceType } from './Substance';
 
-const HALF_LIFE_MS: number = 1000;
+export const ModelProperties = types
+  .model('ModelProperties', {
+    albino: types.optional(types.boolean, false),
+    working_tyr1: types.optional(types.boolean, false),
+    working_myosin_5a: types.optional(types.boolean, true),
+    open_gates: types.optional(types.boolean, false)
+  });
 
-/**
- * An immutable model of an Organism.
- */
-export default abstract class Organism {
-  private name: string;
-  private substanceLevels: { [cellPart in CellPart]: { [substance in Substance]: number} };
-  private substanceDeltas: { [cellPart in CellPart]: { [substance in Substance]: number} };
-  abstract getImageSrc(): string;
-  constructor(name: string, substanceLevels: { [cellPart in CellPart]: { [substance in Substance]: number} }) {
-    this.name = name;
-    this.substanceLevels = substanceLevels;
-    this.substanceDeltas = this.getClearedSubstanceDeltas();
-  }
+export const Organism = types
+  .model('Organism', {
+    id: types.optional(types.identifier(types.string), () => uuid()),
+    organelles: types.map(Organelle),
+    modelProperties: types.optional(ModelProperties, () => ModelProperties.create())
+  })
+  .views(self => ({
+    getImageSrc() {
+      return self.id === 'Field Mouse' ? 'assets/sandrat-light.png' : 'assets/sandrat-dark.png';
+    },
+    getLevelForOrganelleSubstance(organelle: string, substanceType: SubstanceType) {
+      return self.organelles.get(organelle) ? self.organelles.get(organelle).getLevelForSubstance(substanceType) : 0;
+    },
+    getDeltaForOrganelleSubstance(organelle: string, substanceType: SubstanceType) {
+      return self.organelles.get(organelle) ? self.organelles.get(organelle).getDeltaForSubstance(substanceType) : 0;
+    }
+  }))
+  .actions(self => ({
+    incrementOrganelleSubstance(organelleType: string, substanceType: SubstanceType, amount: number) {
+      let organelle = self.organelles.get(organelleType);
+      if (organelle) {
+        organelle.incrementSubstance(substanceType, amount);
+      } else {
+        let newOrganelle = Organelle.create({ type: organelleType });
+        newOrganelle.incrementSubstance(substanceType, amount);
+        self.organelles.set(organelleType, newOrganelle);
+      }
+    }
+  }));
+export type IOrganism = typeof Organism.Type;
 
-  getName(): string { return this.name; }
-  getSubstanceLevels(): { [cellPart in CellPart]: { [substance in Substance]: number} } { 
-    return cloneDeep(this.substanceLevels); 
-  }
-  getSubstanceDeltas(): { [cellPart in CellPart]: { [substance in Substance]: number} } {
-    return cloneDeep(this.substanceDeltas); 
-  }
+export const OrganelleRef = types
```

Comment:
We had discussed a couple of model improvements at the start that I didn't get a chance to make. Specifically, I left the concept of an `OrganelleRef`, rather than adding more traits to the `Organelle` model, and I also left all substance state in non-volatile storage. The second piece might not be a problem though, since eventually we'll be able to modify organism genetics which would change default substance levels.